server/src/main/java/org/elasticsearch/cluster/routing/RoutingTable.java

```diff
@@ -472,19 +474,12 @@ public Builder updateNumberOfReplicas(final int numberOfReplicas, final String[]
                 for (IndexShardRoutingTable indexShardRoutingTable : indexRoutingTable) {
                     builder.addIndexShard(indexShardRoutingTable);
                 }
-                if (currentNumberOfReplicas < numberOfReplicas) {
-                    // now, add "empty" ones
-                    for (int i = 0; i < (numberOfReplicas - currentNumberOfReplicas); i++) {
+                int delta = Math.abs(numberOfReplicas - currentNumberOfReplicas);
+                for (int i = 0; i < delta; i++) {
+                    if (currentNumberOfReplicas < numberOfReplicas) {
                         builder.addReplica();
-                    }
-                } else if (currentNumberOfReplicas > numberOfReplicas) {
-                    int delta = currentNumberOfReplicas - numberOfReplicas;
-                    if (delta <= 0) {
```

Comment:
Maybe just remove this condition as it's obviously always `false` and keep the rest the same (maybe also inline `delta` like we have in the "add" case). Seems to me that makes the logic easiest to digest if we want to change it?

Reply:
@original-brownbear Thanks for checking, I think we could pre-compute the delta and seperate the add and remove operation. Please help to review the update.

Reply:
I'm with @original-brownbear here, using `Math.abs(expr)` and then branching on the sign of `expr` looks weird to me. Let's avoid the use of `Math#abs` (and therefore the super-broad `@SuppressForbidden` annotation).

Reply:
OK. I removed `abs` method. Thank you @DaveCTurner .